pkgs/applications/editors/emacs/generic.nix

```
@@ -67,7 +67,28 @@ let emacs = (if withMacport then llvmPackages_6.stdenv else stdenv).mkDerivation
   pname = pname + lib.optionalString ( !withX && !withNS && !withMacport && !withGTK2 && !withGTK3 ) "-nox";
   inherit version;
 
-  patches = patches fetchpatch;
+  patches = patches fetchpatch ++
+            lib.optional nativeComp
+              (substituteAll {
+                src = let majorVersionStr = lib.versions.major version;
+                          majorVersionInt = lib.toInt majorVersionStr;
+                      in
+                        if majorVersionInt < 29 || (lib.stringLength majorVersionStr == 8 && majorVersionStr < "20221003")
```

I think you can use:
```suggestion
                        if lib.versionOlder version "29"  || lib.versionOlder version "20221003.0"
```

---

OK, I am now using `versionOlder`. I think the `stringLength` check is still needed though.